opencensus/trace/propagation/grpc_trace_bin.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef OPENCENSUS_TRACE_PROPAGATION_GRPC_TRACE_BIN_H_
+#define OPENCENSUS_TRACE_PROPAGATION_GRPC_TRACE_BIN_H_
+
+#include <string>
+
+#include "absl/strings/string_view.h"
+#include "opencensus/trace/span_context.h"
+
+namespace opencensus {
+namespace trace {
+namespace propagation {
+
+// Parses the value of the binary grpc-trace-bin header, returning a
+// SpanContext. If parsing fails, IsValid will be false.
+//
+// Example value, hex encoded:
+//   00                               (version)
+//   00                               (trace_id field)
+//   12345678901234567890123456789012 (trace_id)
+//   01                               (span_id field)
+//   0000000000003039                 (span_id)
+//   02                               (trace_options field)
+//   01                               (options: enabled)
+//
+// See also:
+// https://github.com/census-instrumentation/opencensus-specs/blob/master/encodings/BinaryEncoding.md
+SpanContext FromGRPCTraceBinHeader(absl::string_view header);
+
+// Returns a value for the grpc-trace-bin header.
+std::string ToGRPCTraceBinHeader(const SpanContext& ctx);
```

Comment:
Do we want to also offer a version that outputs to a preallocated buffer? This will allow to easily change the grpc plugin.

Reply:
Done:
```
Benchmark                            Time             CPU   Iterations
----------------------------------------------------------------------
BM_FromGRPCTraceBin               6.56 ns         6.56 ns     83790608
BM_FromGRPCTraceBin_Invalid       1.83 ns         1.83 ns    382096612
BM_ToGRPCTraceBin                 24.7 ns         24.7 ns     28230790
BM_ToGRPCTraceBin_InPlace         5.28 ns         5.28 ns    132377935 <---
```